scripts/entrypoint.sh

```diff
@@ -8,7 +8,11 @@ REAL_USER_ID=${ODK_USER_ID:-1000}
 REAL_GROUP_ID=${ODK_GROUP_ID:-1000}
 
 groupadd -o -g $REAL_GROUP_ID odkuser
-useradd -o -s /bin/bash -u $REAL_USER_ID -g $REAL_GROUP_ID -c "ODK User" -m odkuser
+useradd -o -s /bin/bash -u $REAL_USER_ID -g $REAL_GROUP_ID -c "ODK User" -M odkuser
+mkdir -p /home/odkuser
+chown odkuser:odkuser /home/odkuser
+[ -d /home/odkuser/.data ] && chown odkuser:odkuser /home/odkuser/.data
```

Comment:
I will say something super stupid now, please dismiss gently without too much prejudice:

If we bind the cache dir as volume to the container run, can the chmod change the user permissions on the cache directory on the host in a way that may adversely affect accessing the cache outside the container? 

Sorry, you deserve more expert reviewers.

Reply:
> can the chmod change the user permissions on the cache directory on the host

Not at all (but it’s not a stupid question).

When a directory from the host (let’s say `/home/damien/.data/oaklib`) is bound to a directory in the container (let’s say `/home/odkuser/.data/oaklib`), only the _last_ component of the host directory path is visible from the container (in this case, the `oaklib` directory).

Basically, the `oaklib` directory is the ”cut-off” point between the container and the host. Anything above it only exists in the container, while everything below it comes straight from the host:

```
/home/odkuser/.data/oaklib/something/in/my/cache
\_________________/ \__________________________/
  ONLY EXISTS IN      EXISTS IN THE HOST
  THE CONTAINER
```

Another way to put it is that, just because `/home/damien/.data/oaklib` on the host is bound to `/home/odkuser/.data/oaklib` in the container, it doesn’t mean that `/home/damien/.data` is bound to `/home/odkuser/.data`.

So, any `chmod` made on `/home/odkuser` or `/home/odkuser/.data` will have no effect on `/home/damien/.data/oaklib`.

Of note, this concern about not “adversely affecting the cache outside of the container” is precisely the reason for calling `chmod` twice, first on `/home/odkuser`, then on `/home/odkuser/.data`. We could instead have called `chmod` in recursive mode on `/home/odkuser` (`chmod -R odkuser:odkuser /home/odkuser`), which would have automatically taken care of `/home/odkuser/.data`. **But** this would also have impacted any files and directories below the “cut-off” point (files and directories from the host), which we want to avoid.

Reply:
Awesome this makes total sense! Thank you!